Added Doubly linked list functionality.

Akshat1Nar opened the pull request.

Akshat1Nar committed:
Added Doubly linked list functionality

Akshat1Nar commented:
> _No description provided._

I have added basic funtionlity of linked list and also added some of your functions, **Reversing a linked list** , **Merge sort** , with a lot of basic functionality.

If you add my pull request I am also willing to update the README.md

rishitc commented:
Hi,
Really glad to hear, you are interested in our repository 👍 . Can you make the code interactive, by adding the main function to test if the code works as expected? You can comment it out later.

Also in your code with respect to the ```void swap(int *A, int *B)``` function, can you please change it, to swap two nodes using pointers instead, as that is more efficient, in all general cases as well, as compared to swapping the values of the variables, which may not be as efficient in case of a large structure type.

Thank You.

Akshat1Nar committed:
updated file as per your requirements

Akshat1Nar commented:
Hello, Sorry for the delay.
I updated a swap function that swaps the pointers instead of actual data and also added a main function to test some of the functionality of the program. I had to make some changes to make the main function running and now the thing works without creating any problems, Just check it out and tell me if it was my first contribution to the open-source!


Also, tell me if you want me to turn that readme into the semi-instruction manual. ;)
